kernelci/cli/event.py

```diff
@@ -0,0 +1,83 @@
+# SPDX-License-Identifier: LGPL-2.1-or-later
+#
+# Copyright (C) 2023 Collabora Limited
+# Author: Guillaume Tucker <guillaume.tucker@collabora.com>
+# Author: Jeny Sadadia <jeny.sadadia@collabora.com>
+
+"""Tool to interact with the Pub/Sub interface and message queues"""
+
+import sys
+import json
+
+import click
+
+import kernelci.api
+import kernelci.api.helper
+import kernelci.config
+from . import Args, kci
+
+
+@kci.group(name='event')
+def kci_event():
+    """Interact with Pub/Sub and message queue events"""
+
+
+@kci_event.command(secrets=True)
+@click.argument('channel')
+@Args.config
+@Args.api
+def subscribe(config, api, channel, secrets):
+    """Subscribe to a Pub/Sub channel"""
+    configs = kernelci.config.load(config)
+    api_config = configs['api'][api]
+    api = kernelci.api.get_api(api_config, secrets.api.token)
+    sub_id = api.subscribe(channel)
+    click.echo(sub_id)
+
+
+@kci_event.command(secrets=True)
+@click.argument('sub_id')
+@Args.config
+@Args.api
+def unsubscribe(config, api, sub_id, secrets):
+    """Unsubscribe from a Pub/Sub channel"""
+    configs = kernelci.config.load(config)
+    api_config = configs['api'][api]
+    api = kernelci.api.get_api(api_config, secrets.api.token)
+    api.unsubscribe(sub_id)
+
+
+@kci_event.command(secrets=True)
+@click.option('--is-json', help="Parse input data as JSON", is_flag=True)
+@Args.config
+@Args.api
+@click.argument('channel')
+def send(config, api, is_json, channel, secrets):
+    """Read some data on stdin and send it as an event on a channel"""
+    configs = kernelci.config.load(config)
+    api_config = configs['api'][api]
+    api = kernelci.api.get_api(api_config, secrets.api.token)
+    data = sys.stdin.read()
```

Comment:
By the way, it's also possible to have a File argument with Click.  When the file value is `-` it will read from stdin.  This works, but maybe we should consider using File arguments in all the commands.  Here's the implementation I made a couple of weeks ago:
```python
@kci_node.command(secrets=True)
@click.argument('input_file', type=click.File('r'))
@Args.config
@Args.api
@Args.indent
@catch_http_error
def submit(input_file, config, api, indent, secrets):
    data = json.load(input_file)
    api = _get_api(config, api, secrets)
    func = api.node.update if 'id' in data else api.node.add
    node = func(data)
    click.echo(json.dumps(node, indent=indent or None))
```

Reply:
This can be included in a follow-up PR.